lighthouse-cli/bin.ts

```diff
@@ -272,82 +255,92 @@ function handleError(err: LighthouseError) {
-            `${resolvedPath}.report.${typeToExtension(flags.output)}`;
-        return Printer.write(results, flags.output, outputPath).then(results => {
-          if (flags.output === Printer.OutputMode[Printer.OutputMode.html] ||
-              flags.output === Printer.OutputMode[Printer.OutputMode.domhtml]) {
-            if (flags.view) {
-              opn(outputPath, {wait: false});
-            } else {
-              log.log('CLI', 'Protip: Run lighthouse with `--view` to immediately open the HTML report in your browser');
-            }
+  const typeToExtension = (type: string) => type === 'domhtml' ? 'dom.html' : type;
+  return promise.then(_ => {
+    if (Array.isArray(flags.output)) {
+      return flags.output.reduce((innerPromise, outputType) => {
+        const outputPath = `${resolvedPath}.report.${typeToExtension(outputType)}`;
+        return innerPromise.then((_: Results) => Printer.write(results, outputType, outputPath));
+      }, Promise.resolve(results));
+    } else {
+      const outputPath =
+          flags.outputPath || `${resolvedPath}.report.${typeToExtension(flags.output)}`;
+      return Printer.write(results, flags.output, outputPath).then(results => {
+        if (flags.output === Printer.OutputMode[Printer.OutputMode.html] ||
+            flags.output === Printer.OutputMode[Printer.OutputMode.domhtml]) {
+          if (flags.view) {
+            opn(outputPath, {wait: false});
+          } else {
+            log.log(
+                'CLI',
+                'Protip: Run lighthouse with `--view` to immediately open the HTML report in your browser');
           }
+        }
 
-          return results;
-        });
-      }
-    });
-}
-
-export async function runLighthouse(url: string,
-                       flags: {port: number, skipAutolaunch: boolean, selectChrome: boolean, output: any,
-                         outputPath: string, interactive: boolean, saveArtifacts: boolean, saveAssets: boolean
-                         chromeFlags: string, maxWaitForLoad: number, view: boolean},
-                       config: Object | null): Promise<{}|void> {
-
-  let chromeLauncher: ChromeLauncher | undefined = undefined;
-
-  try {
-    await initPort(flags)
-    const chromeLauncher = await getDebuggableChrome(flags)
-    const results = await lighthouse(url, flags, config);
-
-    const artifacts = results.artifacts;
-    delete results.artifacts;
-
-    await saveResults(results, artifacts!, flags);
-    if (flags.interactive) {
-      await performanceXServer.hostExperiment({url, flags, config}, results);
-    }
-
-    return await chromeLauncher.kill();
-  } catch (err) {
-    if (typeof chromeLauncher !== 'undefined') {
-      await chromeLauncher!.kill();
+        return results;
+      });
     }
-
-    return handleError(err);
-  }
- }
+  });
+}
 
 export function run() {
   return runLighthouse(url, cliFlags, config);
 }
+
+export async function runLighthouse(
```

Comment:
whyd runLighthouse position move? 

would prefer to keep order of `run` and `runLighthouse` the same for a cleaner diff/blame